src/main/java/prefs/UpdateConfig.java

```diff
@@ -0,0 +1,35 @@
+package prefs;
+
+import util.Version;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Contains configuration information for updating
+ */
+public class UpdateConfig {
+
+    private boolean lastUpdateDownloadStatus;
+    private List<Version> listOfVersionsPreviouslyDownloaded; // NOPMD
```

Comment:
What's the warning that's being suppressed?

Also, don't think the name needs the `listOf` prefix. Types are metadata/implementation details.